internal/controller/operatorconfigmap_controller.go

```diff
@@ -393,13 +393,13 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI() error {
 }
 
 func (c *OperatorConfigMapReconciler) deletionPhase() error {
-	claimsList := &v1alpha1.StorageClaimList{}
-	if err := c.list(claimsList, client.Limit(1)); err != nil {
-		c.log.Error(err, "unable to verify StorageClaims presence prior to removal of CSI resources")
+	clientsList := &v1alpha1.StorageClientList{}
+	if err := c.list(clientsList, client.Limit(1)); err != nil {
+		c.log.Error(err, "unable to verify StorageClients presence prior to removal of CSI resources")
 		return err
-	} else if len(claimsList.Items) != 0 {
-		err = fmt.Errorf("failed to clean up resources: storage claims are present on the cluster")
-		c.log.Error(err, "Waiting for all storageClaims to be deleted.")
+	} else if len(clientsList.Items) != 0 {
+		err = fmt.Errorf("failed to clean up resources: storage clients are present on the cluster")
+		c.log.Error(err, "Waiting for all storageClients to be deleted.")
```

Comment:
Why is that needed? The reason we where looking into the existence of claim was that claims themselves where protected from deletion until all PV/PVCs are gone. Are we now giving similar guarantees with the clients? And if so where is the change?

Reply:
yes, all the code was mode to storageclient controller, maybe you need to expand on the hidden files in github. for ex https://github.com/red-hat-storage/ocs-client-operator/blob/e701fb8129fd86e36538b94348d3cc7309b6a1e0/internal/controller/storageclient_controller.go#L829